mesonbuild/build.py

```diff
@@ -1987,6 +1987,10 @@ def post_init(self) -> None:
             # Don't let configuration proceed with a non-static crate type
             elif self.rust_crate_type not in ['rlib', 'staticlib']:
                 raise InvalidArguments(f'Crate type "{self.rust_crate_type}" invalid for static libraries; must be "rlib" or "staticlib"')
+            # See https://github.com/rust-lang/rust/issues/110460
+            if self.rust_crate_type == 'rlib' and any(c in self.name for c in ['-', ' ', '.']):
+                raise InvalidArguments('Rust crate types "dylib" and "proc-macro" do not allow spaces, periods or dashes in the library name '
```

Comment:
```suggestion
                raise InvalidArguments('Rust crate type "rlib" does not allow spaces, periods or dashes in the library name '
```

Reply:
So much back and forth :) I don't know how often I have fixed this already in the meantime. Thanks for finding it again.